templates/default/vlan/vlanmodify-dialog.html

```diff
@@ -12,26 +12,39 @@
                 </tr>
                 <tr>
                     <td>
-                        {icon name="vlan" label="VLAN ID:"}
+                        {icon name="vlan" label="VLAN ID:" class="fa-fw"}
                     </td>
                     <td>
                         <input type="text" name="addvlan[vlanid]" value="{$vlaninfo.vlanid}" required>
                     </td>
                 </tr>
                 <tr>
                     <td class="nobr">
-                        {icon name="description" label="VLAN Description"}
+                        {icon name="description" label="VLAN Description" class="fa-fw"}
                     </td>
                     <td>
                         <input type="text" name="addvlan[description]" value="{$vlaninfo.description}">
                     </td>
                 </tr>
                 <tr>
                     <td>
-                        {icon name="customer" label="Customer"}
+                        {icon name="customer" label="Customer" class="fa-fw"}
                     </td>
                     <td>
-                        {customerlist form="addvlan" selected="{$vlaninfo.customerid}" inputname="addvlan[customerid]"}
+                        {customerlist form="addvlan" selected="{$vlaninfo.customerid}" inputname="addvlan[customerid]" version=2}
+                    </td>
+                </tr>
+                <tr>
+                    <td class="nobr">
+                        {icon name="netnode" label="Network Node" class="fa-fw"}
+                    </td>
+                    <td>
+                        <select name="addvlan[netnodeid]">
+                            <option value="">{trans("- none -")}</option>
+                            {foreach $netnodelist as $nn}
+                                <option value="{$nn.id}" {if $nn.id == $vlaninfo.netnodeid} selected{/if}>{$nn.name} ({$nn.id})</option>
+                            {/foreach}
```

Comment:
Jak już grzebiesz przy węzłach to warto sprawdzić czy jako nazwy węzła nie da się czasem wpisać:
`<script>alert('debian jest do kitu');</script>` i zobaczyć czy nie będziemy mieli również tu XSS-a ;-)

Reply:
Uważam, że rozwiązanie powinno być systemowe na backendzie a obecne dane w bazie można przeszperać pod kątem obecności "<script". 

Mimo to dorzucam na Twoją prośbę również:
https://github.com/lmsgit/lms/pull/1890/commits/fad02ec105c04fda253134652b69963bb47e9626

Reply:
Systemowe tzn. jakie?
Mamy już w html smarty modyfikator `|escape`.
W backendzie mamy `Utils::removeInsercureHtml($value)`.
Jeśli ma odbywać się automatycznie to trzeba by zdefiniować zasady dla jakich pól jaki zbiór znaków dozwalamy we frontendzie, albo wszystko puszczać przez Utils::removeInsecureHtml($value) - choć to drugie spowoduje niepotrzebne mulenie systemu w dłuższej perspektywie. Nazwa komputer ma np. dozwolony zbiór znaków, więc jest to sensownie zabezpieczone, choć nieidealnie, bo jak ktoś dopuści znaki `<'`, '>` i parę innych to może stać się groźnie.
Moim zdaniem nie ma tu nic specjalnie do roboty w ogólności, lecz trzeba robić od przypadku do przypadku i albo dla danego pola sztywno ograniczać zestaw znaków, albo robić Utils::removeInsecureHtml(), albo robić we frontendzie `|escape`. We frontendzie jest o tyle dobre, że nie narzucasz żadnych ograniczeń na ciąg znaków i nic przez przypadek nie wytniesz.

Co do samego XSS - raczej nie rozumiesz w czym rzecz, bo to nie jest problem tylko z `<script>...</script>`, które jest jednym z przypadków XSS. Wyobraż sobie, że masz w HTML:
```html
<input type="text" value="{$value}">
```
a `$value` z backendu ma wartość:
```
jakaś wartość" onmouseover="wyślij_ciastko_bieżącego_użytkownika_na_zdalny_serwer();
```
Wiesz co to spowoduje? :)